crates/cairo-lang-casm/src/hints/mod.rs

```diff
@@ -831,11 +831,27 @@ impl Display for Hint {
             Hint::DebugPrint { start, end } => writedoc!(
                 f,
                 "
-
                     start = {}
                     end = {}
-                    for i in range(start, end):
-                        print(memory[i])
+                    while start != end:
+                        value = memory[start]
+                        str_value = None
+                        try:
+                            non_zero_byte_encountered = False
+                            str_value_stripped = ''
+                            for bt in value.to_bytes(length=31, byteorder='big'):
+                                if not non_zero_byte_encountered and bt == 0:
+                                    continue
+                                if bt != 0:
+                                    non_zero_byte_encountered = True
+                                str_value_stripped += chr(bt)
+                            str_value = str_value_stripped
+                        except BaseException as e:
+                            print('string conversion failed: ' + str(e))
+                        original_value_msg = 'original value: [' + str(value) + ']'
```

Comment:
```suggestion
                        original_value_msg = 'original value: [' + hex(value) + ']'
```

This way we can differentiate bytes visually without converting it to hex by hand. For example printing 'hello' this way results in:

`"original value: [0x68656c6c6f] converted to a string: [hello]"`

Now a user would be able to see a byte corresponding to every character (h = 0x68, e = 0x65 etc.).|
Otherwise we get int('0x68656c6c6f', 16) = 448378203247 - not very user friendly imo.

Counterargument: in cairo0 `serialize_word` doesn't convert to hex.

WDYT?

Reply:
cc @MaksymilianDemitraszek @Arcticae @cptartur

Reply:
I would leave decimal

Reply:
I think both hex and decimal values are equally not very useful. Don't think many users will be handling their errors based on these raw values - we can use any representation imo.

Reply:
> I think both hex and decimal values are equally not very useful. Don't think many users will be handling their errors based on these raw values - we can use any representation imo.

It's not only about handling exceptions but about printing out anything that you may need.

I'd lean towards using the decimal format. This is because, if we use print for debugging, it's easier to read decimal values.

> Now a user would be able to see a byte corresponding to every character (h = 0x68, e = 0x65 etc.).
Otherwise we get int('0x68656c6c6f', 16) = 448378203247 - not very user friendly imo.

That's cool but please note that you're going to have this value's string representation anyways, I don't see how the hex form could be useful. Do you have any particular case for that?

For me, there are several cases:
- you want to print a number - decimal is better than hex
- you want to print a boolean - doesn't matter that much
- you want to print a string - there's a string representation printed
- you want to print an array/other collection - doesn't change anything as each element's going to be printed separately